lib/pbench/server/indexing_tarballs.py

```diff
@@ -103,69 +117,71 @@ class Index:
         ErrorCode("GENERIC_ERROR", 12, False, "Unexpected error encountered"),
     )
 
-    def __init__(self, name, options, idxctx, incoming, archive, qdir):
+    def __init__(self, name: str, options: Namespace, idxctx: IdxContext):
 
-        self.options = options
-        _re_idx = "RE-" if options.re_index else ""
+        # pbench-index command options
+        self.options: Namespace = options
         if options.index_tool_data:
             # The link source and destination for the operation of this script
             # when it only indexes tool data.
-            self.linksrc = "TO-INDEX-TOOL"
-            self.linkdest = "INDEXED"
+            self.operation = Operation.INDEX_TOOL
+            self.enabled = None
         else:
             # The link source and destination for the operation of this script
             # when it indexes run, table-of-contents, and result data.
-            self.linksrc = f"TO-{_re_idx}INDEX"
-            self.linkdest = "TO-INDEX-TOOL"
-        # We only ever use a symlink'd error destination for indexing
-        # problems.
-        self.linkerrdest = "WONT-INDEX"
-        self.idxctx = idxctx
-        self.incoming = incoming
-        self.name = name
-        self.archive = archive
-        self.qdir = qdir
+            self.operation = Operation.RE_INDEX if options.re_index else Operation.INDEX
+            self.enabled = [Operation.INDEX_TOOL]
+
+        # indexing context
+        self.idxctx: IdxContext = idxctx
+
+        # Index context name
+        self.name: str = name
 
-    def collect_tb(self):
-        """Collect tarballs that needs indexing"""
+        # An Elasticsearch status Report object
+        self.report: Optional[Report] = None
 
-        # find -L $ARCHIVE/*/$linksrc -name '*.tar.xz' -printf "%s\t%p\n" 2>/dev/null | sort -n > $list
-        tarballs = []
+        # We'll use a FileTree in order to find the INCOMING directory for the
+        # unpacked tarballs.
+        self.filetree: FileTree = FileTree(idxctx.config, idxctx.logger)
+
+        # Manage synchronization between components
+        self.sync: Sync = Sync(idxctx.logger, "index")  # Build a sync object
+
+    def collect_tb(self) -> Tuple[int, List[TarballData]]:
+        """Collect tarballs that need indexing
+
+        This looks for Datasets marked to require indexing, and returns a tuple
+        consisting of an error number from the 'error_code' attribute and a
+        list of tarball descriptions.
+
+        Returns:
+            A list of tarball descriptions sorted by tarball size.
```

Comment:
I think the text starting with `returns` at line 154 is correct, while the text under `Returns` at line 158 is not so much.  I think the text from the end of line 154 should be moved to and merged with the text at line 159, so that truth is restored to the universe.